src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -405,13 +427,24 @@ protected IEnumerable<Symbol> GetUnsafeAddressTaken()
 
         #region Tracking reads/writes of variables for warnings
 
-        protected virtual void NoteRead(Symbol variable, ParameterSymbol rangeVariableUnderlyingParameter = null)
+        protected virtual void NoteRead(
+            Symbol variable,
+            ParameterSymbol rangeVariableUnderlyingParameter = null)
         {
             var local = variable as LocalSymbol;
             if ((object)local != null)
             {
-                _usedVariables.Add(local);
+                if (IsCapturedInLocalFunction(local,
+                                              rangeVariableUnderlyingParameter))
+                {
+                    NoteReadInLocalFunction(local, local);
```

Comment:
>NoteReadInLocalFunction [](start = 20, length = 23)

I don't see that you have enough information to infer that it was *read inside* a local function. It may merely have been written inside a local function.